web/src/components/PixiMap.tsx

```diff
@@ -147,38 +171,190 @@ const RequiresInitLayer = <T,>({
+    <Stack spacing={0} position="relative" height={height}>
+      {showCitySelectionDropDown ? (
+        <Flex
+          position="absolute"
+          zIndex={5}
+          top={selectionDropdownTop}
+          left={selectionDropdownLeft}
+          background={colorMode === "light" ? "#553ACF" : "#CCC3F9"}
+          color={colorMode === "light" ? "white" : "black"}
+          boxShadow="1px 6px 6px grey"
+          width="-webkit-fit-content"
+          justifyContent="space-between"
+          gap={1}
+          padding="0px 2px 0px 10px"
+          borderRadius="6px"
+          alignItems="center"
+        >
+          <Box display="inline">
+            <Text display="inline">Map view</Text>
+          </Box>
+          <Center
+            height="25px"
+            margin="2px 0px 2px 5px"
+            background={colorMode === "light" ? "white" : "black"}
+          >
+            <Divider
+              color={colorMode === "light" ? "black" : "white"}
+              orientation="vertical"
+            />
+          </Center>
+          <Tooltip
+            isDisabled={
+              !(selectedCities.length <= 0 || isUpdating || !connected)
+            }
+            label={dropdownDisabledMsg}
+          >
+            <Box display="inline">
+              <Select
+                options={selectedCities.map((c) => ({
+                  label: c.name,
+                  value: c,
+                }))}
+                value={selectionValue}
+                onChange={(e) => setLastSelectedCityId(() => e?.value.id || -1)}
+                isDisabled={
+                  selectedCities.length <= 0 || isUpdating || !connected
+                }
+                styles={{
+                  input: (props) => ({
+                    ...props,
+                    background: "transparent",
+                    cursor: "pointer",
+                  }),
+                  placeholder: (props) => ({
+                    ...props,
+                    color: colorMode === "light" ? "white" : "black",
+                  }),
+                  option: (props) => ({
+                    ...props,
+                    background: "white",
+                    cursor: "pointer",
+                    color: "#4C4A57",
+                  }),
+                  dropdownIndicator: (props) => ({
+                    ...props,
+                    color: colorMode === "light" ? "white" : "black",
+                  }),
+                  indicatorSeparator: (props) => ({
+                    ...props,
+                    display: "none",
+                  }),
+                  control: (props) => ({
+                    ...props,
+                    background: "transparent",
+                    border: "none",
+                  }),
+                  singleValue: (props) => ({
+                    ...props,
+                    color: colorMode === "light" ? "white" : "black",
+                  }),
+                }}
+              />
```

Comment:
This whole dropdown should be a separate component. It is very difficult to parse what is happening here.

Besides a separate component, somethings should be calculated separately to better understand what is going on. For example, the `styles` could be a variable calculated before the return (you could also calculate the white/black color once and reuse it). `isDisabled` could also be a variable determined beforehand.

Reply:
I have rearranged this section into smaller component functions should be neat now